### ISSUE TITLE
Cutting a new release with compiler fixes

### ISSUE THREAD

**Issue by zealot128** [vhyza/language_detection#2]
Hi there!

I've found this repo, which is a really nice alternative to the WhatsLanguage gem. But installing it did not work first, but produce a Compiler error.

Then I've tried to install from git/source, and that worked, as it seems to be the error was already fixed in one of the recent commits.

Could you make a new release that includes those fixes?

**Comment by vhyza** [comment]
Hi @zealot128,

Wow. Blast from the past 😅. I did not use this gem for a decade or something 🙂. So I do not really remember the status of it.

I'm now traveling, so I'm more or less offline, but I'll try to take a look at it during the Christmas...

Feel free to ping me if there is no progress in the beginning of the year...

**Comment by zealot128** [comment]
> 

No worries, I just found that Gem when looking for an alternative for WhatLanguage, which does not work very well. I can use the Gem as a Git source in the meantime. Works really great!

Happy holidays and safe travel :)

**Comment by vhyza** [comment]
Thank you @zealot128. You too :-)

I tried to install the gem (`v0.0.2`) via

```bash
gem install language_detection
```

using Ruby 2.6, 2.7 and 3.0 without any error. I had problems with Ruby 3.1 as I couldn't install `ffi` gem. After updating `ffi` at least to the v1.12 it seems working.

So I refactored gem dependencies and tried the new version (`v0.1.0`) with ruby `2.5`, `2.6`, `2.7`, `3.0`, `3.1` and `3.2preview2`. As the Ruby `2.5` is now more than year EOL I also set the supported Ruby versions in gemspec to `gem.required_ruby_version = [ ">= 2.5.0", "< 3.3.0" ]`

I hope I did not break some compatibility as it is new release after more than decade 😅

Also when I was looking if I can upgrade the CLD to latest version (https://github.com/google/cld3) and I found that there is already rubygem implementing it - https://github.com/akihikodaki/cld3-ruby. So maybe it can be alternative as I'm not sure I'm able to update the CLD...
